### ISSUE TITLE
[0.8 and before] Hogwart's Legacy still consistently crashes after a few minutes of playing

### ISSUE THREAD

**Issue by Dravic** [Nukem9/dlssg-to-fsr3#235]
2080 Ti

Just flying around on a broom or walking around Hogsmeade is enough to get Hogwart's Legacy to crash.

The only way to prevent those crashes is to get rid of the mod completely, then I can fly around the broom or walk around Hogsmeade for a long time and I do not crash.

Tested both DLSSTweaks version (0.8) and non-DLSSTweaks version (before), same result.

**Comment by fatezerotwoxyz** [comment]
Same, I think this has something to do with VRAM limit, when the VRAM limit is reached, using this mod will cause crashes in many Unreal Engine games, hope this can be fixed.

For now I add r.Streaming.PoolSize in the game's Engine.ini file to limit the VRAM usage and it can be played longer.

**Comment by Erebus1245** [comment]
Yeah at least stating your Hardware would helpful not to mention [the game is an absolute ram/vram hog](https://en.overclocking.com/hogwarts-legacy-worse-than-chrome-for-ram-vram/) especially with rt and framegen on top of that.

**Comment by Dravic** [comment]
> Yeah at least stating your Hardware would helpful

You would never believe what the first line of my thread contains.

That's right! My GPU!

>[The game is an absolute ram/vram hog](https://en.overclocking.com/hogwarts-legacy-worse-than-chrome-for-ram-vram/) especially with rt and framegen on top of that.

Cool but it doesn't crash the same way without Frame Generation mod installed, leading me to believe perhaps there's some issue here.

Could just be a coincidence that the game crashed within minutes repeatedly every time I tried it - and I tried seven or eight times - with Frame Generation installed, but has not crashed for half an hour without Frame Generation mod installed.

**Comment by Erebus1245** [comment]
> > Yeah at least stating your Hardware would helpful
> 
> You would never believe what the first line of my thread contains.
> 
> That's right! My GPU!
> 
> > [The game is an absolute ram/vram hog](https://en.overclocking.com/hogwarts-legacy-worse-than-chrome-for-ram-vram/) especially with rt and framegen on top of that.
> 
> Cool but it doesn't crash the same way without Frame Generation mod installed, leading me to believe perhaps there's some issue here.
> 
> Could just be a coincidence that the game crashed within minutes repeatedly every time I tried it - and I tried seven or eight times - with Frame Generation installed, but has not crashed for half an hour without Frame Generation mod installed.
> 
> I understand that this could just be Hogwart's Legacy quirk, but after a bunch of crashes in a row I thought it is a very consistent issue worth reporting.

Framegen also uses more vram so who knows and i meant your whole system since the game also uses way more system ram then average.

**Comment by Erebus1245** [comment]
I'd say at least monitor you vram/ram usage and try the game without rt.

**Comment by Dravic** [comment]
>i meant your whole system since the game also uses way more system ram then average.

I have 32GB RAM, I am good on that front.

>I'd say at least monitor you vram/ram usage and try the game without rt.

I'll look into it when I have time to test again!

**Comment by MJ12MJ** [comment]
Yes, still crashing.
This is the log file and looks like something is off with the improved texture format, but the game did indeed runned for longer than the previous file.

"LowLevelFatalError [File:Unknown] [Line: 772] \npDevice->GetTextureAllocator().AllocateTexture(TextureDesc, ClearValue, Format, *OutTexture2D, InitialState, Name) failed \n at W:/Engine/Source/Runtime/D3D12RHI/Private/D3D12Texture.cpp:821 \n with error DXGI_ERROR_DEVICE_REMOVED, \n Size=1024x1024x1 Format=DXGI_FORMAT_BC5_UNORM(0x00000053), NumMips=11, Flags=\n\n\n"

**Comment by MJ12MJ** [comment]
> I'd say at least monitor you vram/ram usage and try the game without rt.

It's not related to VRAM/RAM, this is a common issue with the mod.

**Comment by fatezerotwoxyz** [comment]
Game does not crash when using the native DLSS FG when hitting VRAM limit, but it does when using this dlssg-to-fsr3 mod in Unreal Engine games, so there are still some issues that need to be fixed.

**Comment by MJ12MJ** [comment]
> Game does not crash when using the native DLSS FG when hitting VRAM limit, but it does when using this dlssg-to-fsr3 mod in Unreal Engine games, so there are still some issues that need to be fixed.

Yes, considering that first of all it would be visible from the system logs if there is a crash due to the video memory, but this usually doesn't happen because once the limit is reached the game switches to RAM memory, obviously with all the resulting defects such as a very low framerate.  
Nvidia has RTX 40 cards with 8GB VRAM that support FG and the latter do not have the same crash problem, so I don't understand why on cards with 10/12 GB of VRAM the game should crash using FSR3. This doesn't happen even when FSR3 is natively supported on other games, like Avatar for example. For this reason I find it out of place as a problem statement, those who are capable should provide their logs in order to help Nukem resolve these crashes. I understand that Hogwarts Legacy had VRAM issues during its debut, but with the eventual patches that has now been resolved.

**Comment by hemantgogia09** [comment]
+1 for crashing even with FG off or on with mod dll (v0.8) loaded it crashes after 30 secs for me. I am trying with latest steam version not the Jack Sparrow version.

Graphics in game that i am trying:
- RT totally off
- DLSS Quality
- Everything is at Ultra except fog, shadow and post processing which are on high

System spec for reference:
Target Res: 4k
CPU: 5600x
GPU: RTX 3080 10GB
RAM: 32 GB duel channel 3600 MHz

**Comment by FlowersPowerz** [comment]
Never had any crash after 4 hours of continuous playing with a 3090ti which have 24 Gb of vram, so I think could be a vram issue

**Comment by fatezerotwoxyz** [comment]
> > Game does not crash when using the native DLSS FG when hitting VRAM limit, but it does when using this dlssg-to-fsr3 mod in Unreal Engine games, so there are still some issues that need to be fixed.
> 
> Yes, considering that first of all it would be visible from the system logs if there is a crash due to the video memory, but this usually doesn't happen because once the limit is reached the game switches to RAM memory, obviously with all the resulting defects such as a very low framerate. Nvidia has RTX 40 cards with 8GB VRAM that support FG and the latter do not have the same crash problem, so I don't understand why on cards with 10/12 GB of VRAM the game should crash using FSR3. This doesn't happen even when FSR3 is natively supported on other games, like Avatar for example. For this reason I find it out of place as a problem statement, those who are capable should provide their logs in order to help Nukem resolve these crashes. I understand that Hogwarts Legacy had VRAM issues during its debut, but with the eventual patches that has now been resolved.

The latest version of Hogwarts Legacy just handles video memory management better, it doesn't mean it won't swap with system RAM when hitting the VRAM limit, I think this bug is specific to Unreal Engine, maybe there is some implementation conflicts with some of the code in this mod,  and not all crash events can be seen in the system logs, unfortunately.

**Comment by MJ12MJ** [comment]
> > > Game does not crash when using the native DLSS FG when hitting VRAM limit, but it does when using this dlssg-to-fsr3 mod in Unreal Engine games, so there are still some issues that need to be fixed.
> > 
> > 
> > Yes, considering that first of all it would be visible from the system logs if there is a crash due to the video memory, but this usually doesn't happen because once the limit is reached the game switches to RAM memory, obviously with all the resulting defects such as a very low framerate. Nvidia has RTX 40 cards with 8GB VRAM that support FG and the latter do not have the same crash problem, so I don't understand why on cards with 10/12 GB of VRAM the game should crash using FSR3. This doesn't happen even when FSR3 is natively supported on other games, like Avatar for example. For this reason I find it out of place as a problem statement, those who are capable should provide their logs in order to help Nukem resolve these crashes. I understand that Hogwarts Legacy had VRAM issues during its debut, but with the eventual patches that has now been resolved.
> 
> The latest version of Hogwarts Legacy just handles video memory management better, it doesn't mean it won't swap with system RAM when hitting the VRAM limit, I think this bug is specific to Unreal Engine, maybe there is some implementation conflicts with some of the code in this mod, and not all crash events can be seen in the system logs, unfortunately.

Nvidia Aftermath reports crash (GPU-DMP) + UE (GAME DMP) + Windows Log (Drivers etc), so you can definitely look for what's happening, people's just don't know where to look. 
this is precisely my thought, I think there is a conflict with the unreal engine as I reported in my previous report.

**Comment by fatezerotwoxyz** [comment]
> > > > Game does not crash when using the native DLSS FG when hitting VRAM limit, but it does when using this dlssg-to-fsr3 mod in Unreal Engine games, so there are still some issues that need to be fixed.
> > > 
> > > 
> > > Yes, considering that first of all it would be visible from the system logs if there is a crash due to the video memory, but this usually doesn't happen because once the limit is reached the game switches to RAM memory, obviously with all the resulting defects such as a very low framerate. Nvidia has RTX 40 cards with 8GB VRAM that support FG and the latter do not have the same crash problem, so I don't understand why on cards with 10/12 GB of VRAM the game should crash using FSR3. This doesn't happen even when FSR3 is natively supported on other games, like Avatar for example. For this reason I find it out of place as a problem statement, those who are capable should provide their logs in order to help Nukem resolve these crashes. I understand that Hogwarts Legacy had VRAM issues during its debut, but with the eventual patches that has now been resolved.
> > 
> > 
> > The latest version of Hogwarts Legacy just handles video memory management better, it doesn't mean it won't swap with system RAM when hitting the VRAM limit, I think this bug is specific to Unreal Engine, maybe there is some implementation conflicts with some of the code in this mod, and not all crash events can be seen in the system logs, unfortunately.
> 
> Nvidia Aftermath reports crash (GPU-DMP) + UE (GAME DMP) + Windows Log (Drivers etc), so you can definitely look for what's happening, people's just don't know where to look. this is precisely my thought, I think there is a conflict with the unreal engine as I reported in my previous report.

View gpudmp files using NVIDIA Nsight Graphics, I'm seeing the same crash info you reported: MMU fault detected during a Read of address, failed to translate the virtual address, so this is indeed a conflict issue specific to Unreal Engine.

**Comment by Marjaperse** [comment]
Same here. Legit copy + 0.8 version of the mod = crash

**Comment by Naberno** [comment]
> Same here. Legit copy + 0.8 version of the mod = crash
> 
> I've gotten only 2 games to work. Cyberpunk and RoboCop

Direction
Where did u put the dlssfg-to-fsr3 file? 
I copied to that path:  \Engine\Plugins\Runtime\Nvidia\Streamline\Binaries\ThirdParty\Win64

**Comment by Dravic** [comment]
> > Same here. Legit copy + 0.8 version of the mod = crash
> > I've gotten only 2 games to work. Cyberpunk and RoboCop
> 
> Direction Where did u put the dlssfg-to-fsr3 file? I copied to that path: \Engine\Plugins\Runtime\Nvidia\Streamline\Binaries\ThirdParty\Win64

For Robocop or Cyberpunk?

For Robocop you want the mod here:
`...\RoboCop Rogue City's directory\Game\Binaries\Win64`

For Cyberpunk 2077 you want the mod here:
`...\Cyberpunk 2077's directory\bin\x64`

**Comment by Naberno** [comment]
> > > Same here. Legit copy + 0.8 version of the mod = crash
> > > I've gotten only 2 games to work. Cyberpunk and RoboCop
> > 
> > 
> > Direction Where did u put the dlssfg-to-fsr3 file? I copied to that path: \Engine\Plugins\Runtime\Nvidia\Streamline\Binaries\ThirdParty\Win64
> 
> For Robocop or Cyberpunk?
> 
> For Robocop you want the mod here: `...\RoboCop Rogue City's directory\Game\Binaries\Win64`
> 
> For Cyberpunk 2077 you want the mod here: `...\Cyberpunk 2077's directory\bin\x64`

The same Hogwarts legacy

**Comment by Vairy02** [comment]
Can someone of you that have the legit copy of the game share the newer nvidia streamline dll here please? Like sl.xxxx.dll

**Comment by Dravic** [comment]
> Can someone of you that have the legit copy of the game share the newer nvidia streamline dll here please? Like sl.xxxx.dll

It made no positive difference for me when I put Cyberpunk 2077's streamline interposer files in Hogwart's Legacy directory, replacing Hogwart Legacy's original files.

**Comment by Vairy02** [comment]
> > Can someone of you that have the legit copy of the game share the newer nvidia streamline dll here please? Like sl.xxxx.dll
> 
> It made no positive difference for me when I put Cyberpunk 2077's streamline interposer files in Hogwart's Legacy directory, replacing Hogwart Legacy's original files. The game still crashes soon enough when playing. I've since reverted to original interposer files.

i know but i can't get the mod to work cause in the cracked version the streamline files are too old, i just want to test if i can get the mod at least to work with the newer streamline from the original copy, i really appreciate if you can share the dll here (common, dlss_g, interposer, reflex)

**Comment by Dravic** [comment]
> > > Can someone of you that have the legit copy of the game share the newer nvidia streamline dll here please? Like sl.xxxx.dll
> > 
> > 
> > It made no positive difference for me when I put Cyberpunk 2077's streamline interposer files in Hogwart's Legacy directory, replacing Hogwart Legacy's original files. The game still crashes soon enough when playing. I've since reverted to original interposer files.
> 
> i know but i can't get the mod to work cause in the cracked version the streamline files are too old, i just want to test if i can get the mod at least to work with the newer streamline from the original copy, i really appreciate if you can share the dll here (common, dlss_g, interposer, reflex)

Winter Steam Sale is happening right now and Hogwart's Legacy is 50% off.

**Comment by Neegger** [comment]
I can run the game at ultra at 1440p with my 3060ti with rt on at 30fps and it still doesn't crash, unless the mod is there, so it is def. the mod, otherwise it would crash even without it because of the 8gb vram

**Comment by Vairy02** [comment]
> > > > Can someone of you that have the legit copy of the game share the newer nvidia streamline dll here please? Like sl.xxxx.dll
> > > 
> > > 
> > > It made no positive difference for me when I put Cyberpunk 2077's streamline interposer files in Hogwart's Legacy directory, replacing Hogwart Legacy's original files. The game still crashes soon enough when playing. I've since reverted to original interposer files.
> > 
> > 
> > i know but i can't get the mod to work cause in the cracked version the streamline files are too old, i just want to test if i can get the mod at least to work with the newer streamline from the original copy, i really appreciate if you can share the dll here (common, dlss_g, interposer, reflex)
> 
> Winter Steam Sale is happening right now and Hogwart's Legacy is 50% off.

i don't want to play the game, i just want to test the mod.

**Comment by Naberno** [comment]
> > > > > Can someone of you that have the legit copy of the game share the newer nvidia streamline dll here please? Like sl.xxxx.dll
> > > > 
> > > > 
> > > > It made no positive difference for me when I put Cyberpunk 2077's streamline interposer files in Hogwart's Legacy directory, replacing Hogwart Legacy's original files. The game still crashes soon enough when playing. I've since reverted to original interposer files.
> > > 
> > > 
> > > i know but i can't get the mod to work cause in the cracked version the streamline files are too old, i just want to test if i can get the mod at least to work with the newer streamline from the original copy, i really appreciate if you can share the dll here (common, dlss_g, interposer, reflex)
> > 
> > 
> > Winter Steam Sale is happening right now and Hogwart's Legacy is 50% off.
> 
> i don't want to play the game, i just want to test the mod.

++++++

**Comment by Dravic** [comment]
> i don't want to play the game, i just want to test the mod.

First of all that's not an excuse, or rather I don't care about what the excuse is.
Do whatever you want, man, but don't talk about it in public.

Second of all, you have a 2 hours of playtime return window within 14 days of purchase on Steam.
That is plenty of time to test the mod. So you could just do that.

**Comment by Vairy02** [comment]
Obviously its not an excuse, i was just telling you why i don't want to buy the game, and respected that you didn't want to share the files and that's ok, i can understand it.

**Comment by RJASSI98** [comment]
I tried it on my pc:

12900KS
2x32GB DDR5 6600Mhz CL32
Aorus Master RTX 3080 10GB

Crashes after loading into the game within 3 seconds. If I load in then pause into the menu there is no crash. Unpause and it crashes. 

Tried it on my brother's pc

10700K
32GB DDR4
RTX 3090 FE

I used the same save file and it didn't crash. Both using the latest drivers. This is with hogwarts. I haven't tried the mod with any other games.

**Comment by RJASSI98** [comment]
So I tried the game at 1440p with an Asus pg27aqdm on my pc RTX 3080 and the game was fine at ultra with HDR and dlss set to quality. When I run the game on my lg C2 4k with HDR the game crashes 3 seconds in. Also when I ran this on my brother's pc that was also on a C2. 

Set my C2 to 1440p and the game worked fine.

**Comment by Nukem9** [comment]
I looked at Hogwarts Legacy yesterday and it's due to VRAM exhaustion. Once Unreal's texture streaming system eats up all available GPU VRAM, the game hangs or immediately crashes to desktop.

I've also confirmed it'll crash with "Frame Generation" set to "Off" when mod DLLs are installed. Even when I remove all FSR 3 code, but keep the DLLs installed in the game folder, it'll continue to crash. I can't fix a broken game.

---

The only workaround I can think of is to reduce the texture streaming budget (`r.Streaming.PoolSize`) by messing with Unreal's configuration settings.

**Comment by RJASSI98** [comment]
> I looked at Hogwarts Legacy yesterday and it's due to VRAM exhaustion. Once Unreal's texture streaming system eats up all available GPU VRAM, the game hangs or immediately crashes to desktop.
> 
> I've also confirmed it'll crash with "Frame Generation" set to "Off" when mod DLLs are installed. Even when I remove all FSR 3 code, but keep the DLLs installed in the game folder, it'll continue to crash. ~I can't fix a broken game.~
> 
> The only workaround I can think of is to reduce the texture streaming budget (`r.Streaming.PoolSize`) by messing with Unreal's configuration settings. Or you could reduce your graphics settings, such as disabling ray tracing.

I will try reducing settings considering everything was set to ultra. Other than that I could see if I could find a way to mess with the unreal config settings. Thanks for testing and letting me know it most probably is a vram issue though. I'm sure the issue only happened to people with 10-12gb vram though anything above or under wouldn't cause crashes from what I read elsewhere. Would be nice if FSR 3 was supported officially on hogwarts though.

**Comment by FlowersPowerz** [comment]
> > I looked at Hogwarts Legacy yesterday and it's due to VRAM exhaustion. Once Unreal's texture streaming system eats up all available GPU VRAM, the game hangs or immediately crashes to desktop.
> > I've also confirmed it'll crash with "Frame Generation" set to "Off" when mod DLLs are installed. Even when I remove all FSR 3 code, but keep the DLLs installed in the game folder, it'll continue to crash. ~I can't fix a broken game.~
> > The only workaround I can think of is to reduce the texture streaming budget (`r.Streaming.PoolSize`) by messing with Unreal's configuration settings. Or you could reduce your graphics settings, such as disabling ray tracing.
> 
> I will try reducing settings considering everything was set to ultra. Other than that I could see if I could find a way to mess with the unreal config settings. Thanks for testing and letting me know it most probably is a vram issue though. I'm sure the issue only happened to people with 10-12gb vram though anything above or under wouldn't cause crashes from what I read elsewhere. Would be nice if FSR 3 was supported officially on hogwarts though.

With official support you can't activate dlss with the fsr3, and you can't activate reflex, so the input lag is atrocious. I think this is a better implementation apart from the vram issue

**Comment by RJASSI98** [comment]
> > > I looked at Hogwarts Legacy yesterday and it's due to VRAM exhaustion. Once Unreal's texture streaming system eats up all available GPU VRAM, the game hangs or immediately crashes to desktop.
> > > I've also confirmed it'll crash with "Frame Generation" set to "Off" when mod DLLs are installed. Even when I remove all FSR 3 code, but keep the DLLs installed in the game folder, it'll continue to crash. ~I can't fix a broken game.~
> > > The only workaround I can think of is to reduce the texture streaming budget (`r.Streaming.PoolSize`) by messing with Unreal's configuration settings. Or you could reduce your graphics settings, such as disabling ray tracing.
> > 
> > 
> > I will try reducing settings considering everything was set to ultra. Other than that I could see if I could find a way to mess with the unreal config settings. Thanks for testing and letting me know it most probably is a vram issue though. I'm sure the issue only happened to people with 10-12gb vram though anything above or under wouldn't cause crashes from what I read elsewhere. Would be nice if FSR 3 was supported officially on hogwarts though.
> 
> With official support you can't activate dlss with the fsr3, and you can't activate reflex, so the input lag is atrocious. I think this is a better implementation apart from the vram issue

Yes, it most definitely is a better implementation given that it doesn't crash. I'm sure it is great when it works.

**Comment by xv0t** [comment]
> With official support you can't activate dlss with the fsr3, and you can't activate reflex, so the input lag is atrocious. I think this is a better implementation apart from the vram issue

Just a reminder that Special K and the newest RTSS beta both are able to inject Reflex on any DX11/12 game and should work with frame gen.
